_posts/2021-04-06-run-minitest-on-github-actions-with-parallel-jobs-using-build-matrix.md

```diff
@@ -0,0 +1,121 @@
+
+on: [push]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    # If you need DB like PostgreSQL, Redis then define service below.
+    # https://github.com/actions/example-services/tree/master/.github/workflows
+    services:
+      postgres:
+        image: postgres:10.8
+        env:
+          POSTGRES_USER: postgres
+          POSTGRES_PASSWORD: ""
+          POSTGRES_DB: postgres
+        ports:
+          - 5432:5432
+        # needed because the postgres container does not provide a healthcheck
+        # tmpfs makes DB faster by using RAM
+        options: >-
+          --mount type=tmpfs,destination=/var/lib/postgresql/data
+          --health-cmd pg_isready
+          --health-interval 10s
+          --health-timeout 5s
+          --health-retries 5
+      redis:
+        image: redis
+        ports:
+          - 6379:6379
+        options: --entrypoint redis-server
+
+    # https://help.github.com/en/articles/workflow-syntax-for-github-actions#jobsjob_idstrategymatrix
+    strategy:
+      fail-fast: false
+      matrix:
+        # Set N number of parallel jobs you want to run tests on.
+        # Use higher number if you have slow tests to split them on more parallel jobs.
+        # Remember to update ci_node_index below to 0..N-1
+        ci_node_total: [8]
+        # set N-1 indexes for parallel jobs
+        # When you run 2 parallel jobs then first job will have index 0, the second job will have index 1 etc
+        ci_node_index: [0, 1, 2, 3, 4, 5, 6, 7]
+
+    env:
+      RAILS_ENV: test
+      PGHOST: localhost
+      PGUSER: postgres
+      # Rails verifies Time Zone in DB is the same as time zone of the Rails app
+      TZ: "Europe/Warsaw"
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Ruby
+        uses: ruby/setup-ruby@v1
+        with:
+          # Not needed with a .ruby-version file
+          ruby-version: 2.7
+          # runs 'bundle install' and caches installed gems automatically
+          bundler-cache: true
+
+      - name: Create DB
+        run: |
+          bin/rails db:prepare
+      - name: Run tests
+        env:
+          KNAPSACK_PRO_TEST_SUITE_TOKEN_MINITEST: ${{ secrets.KNAPSACK_PRO_TEST_SUITE_TOKEN_MINITEST }}
+          KNAPSACK_PRO_CI_NODE_TOTAL: ${{ matrix.ci_node_total }}
+          KNAPSACK_PRO_CI_NODE_INDEX: ${{ matrix.ci_node_index }}
+          KNAPSACK_PRO_FIXED_QUEUE_SPLIT: true
+          KNAPSACK_PRO_LOG_LEVEL: info
+        run: |
+          bundle exec rake knapsack_pro:queue:minitest
+{% endraw %}
+{% endhighlight %}
+
+## Summary
+
+As you can see the slow Minitest test suite is no longer an issue for you. QA, Testers, or Automation Engineers can benefit from improving the CI build speed and allowing their software developers team to deliver products faster. You can learn more at [Knapsack Pro](https://knapsackpro.com/?utm_source=docs_knapsackpro&utm_medium=blog_post&utm_campaign=run-minitest-on-github-actions-with-parallel-jobs-using-build-matrix).
```

Comment:
unsure about the "is no longer an issue for you" part.

Maybe it's better to say something like "doesn't need be an issue for you". Then in the next sentence we could change "can benefit" to "could benefit".

Reply:
@ArturT sorry I made a mistake in the suggestion. This should read "doesn't need _to_ be an issue for you" instead.

Reply:
Ok. Fixed in https://github.com/KnapsackPro/docs.knapsackpro.com/commit/36dc0e6928e1d8e3887a14269d333d2a137ea609